cloudsplaining/command/scan_multi_account.py

```diff
@@ -0,0 +1,205 @@
+)
+@click.option(
+    "--config-file",
+    "-c",
+    type=click.Path(exists=True),
+    required=True,
+    help="Path of the multi-account config file",
+)
+@click.option(
+    "--profile",
+    "-p",
+    type=str,
+    required=False,
+    help="Specify the AWS IAM profile.",
+    envvar="AWS_PROFILE"
+)
+@click.option(
+    "--role-name",
+    "-r",
+    type=str,
+    required=True,
+    help="The name of the IAM role to assume in target accounts. Must be the same name in all target accounts."
+)
+@click.option(
+    "--exclusions-file",
+    help="A yaml file containing a list of policy names to exclude from the scan.",
+    type=click.Path(exists=True),
+    required=False,
+    default=EXCLUSIONS_FILE,
+)
+@optgroup.group(
+    "Output Target Options",
+    cls=RequiredMutuallyExclusiveOptionGroup,
+    help="",
+)
+@optgroup.option(
+    "--output-directory",
+    "-o",
+    "output_directory",
+    required=False,
+    type=click.Path(exists=True),
+    # default=os.getcwd(),
+    help="Output directory. Supply this or --bucket.",
+)
+@optgroup.option(
+    "--bucket",
+    "-b",
+    "save_bucket",
+    type=str,
+    help="The S3 bucket to save the results. Supply this or --output-directory."
+    # TODO: Validate that this
+)
+@optgroup.group(
+    "Other Options",
+    help="",
+)
+@optgroup.option(
+    "--write-data-file",
+    is_flag=True,
+    required=False,
+    default=False,
+    help="Save the cloudsplaining JSON-formatted data results."
+)
+@click.option(
+    "-v",
+    "--verbose",
+    "verbosity",
+    count=True,
+)
+def scan_multi_account(config_file: str, profile: str, role_name: str, exclusions_file: str, output_directory: str, save_bucket: str, write_data_file: bool, verbosity: int):
+    """Scan multiple accounts via AssumeRole"""
+    set_log_level(verbosity)
+
+    # Read the config file from the user
+    multi_account_config = MultiAccountConfig(config_file=config_file, role_name=role_name)
+
+    # Get the exclusions file
+    exclusions = get_exclusions(exclusions_file=exclusions_file)
+
+    # TODO: Speed improvements? Multithreading? idk.
+    for target_account_name, target_account_id in multi_account_config.accounts.items():
+        results = scan_account(target_account_id=target_account_id, target_role_name=role_name, exclusions=exclusions, profile=profile)
+        html_report = HTMLReport(
+            account_id=target_account_id,
+            account_name=target_account_name,
+            results=results,
+            minimize=True,
+        )
+        rendered_report = html_report.get_html_report()
+        if save_bucket:
+            logger.info("Saving the report to an S3 bucket!")
+            s3 = boto3.resource('s3')
+            # Write the HTML file
+            output_file = f"{target_account_name}.html"
+            s3.Object(save_bucket, output_file).put(ACL='bucket-owner-full-control', Body=rendered_report)
+            # Write the JSON data file
+            if write_data_file:
```

Comment:
❓  Should `write_data_file` be `true` by default? There may need to be a way to override it.

Also does it make sense to use different conditionals for writing the file locally vs to S3? If `output_directory` is present then write to local and if `save_bucket` is present write to S3.

Reply:
Ah, I see what you mean. Like, perhaps we want to save it to both.

I just added that in a recent commit.